src/algebra/lie/abelian.lean

```diff
@@ -139,6 +140,66 @@ begin
     rw [← mem_maximal_trivial_submodule R L M, h], exact lie_submodule.mem_top m, },
 end
 
+variables {R L M N}
+
+/-- `maximal_trivial_submodule` is functorial. -/
+def maximal_trivial_hom (f : M →ₗ⁅R,L⁆ N) :
+  maximal_trivial_submodule R L M →ₗ⁅R,L⁆ maximal_trivial_submodule R L N :=
+{ to_fun    := λ m, ⟨f m, λ x, by
+    { have h := congr_arg f (m.property x),
+      rw [lie_module_hom.map_zero, lie_module_hom.map_lie] at h, exact h, }⟩,
+  map_add'  := λ m n, by simpa,
+  map_smul' := λ t m, by simpa,
+  map_lie'  := λ x m, by simp, }
+
+@[norm_cast, simp] lemma coe_maximal_trivial_hom_apply
+  (f : M →ₗ⁅R,L⁆ N) (m : maximal_trivial_submodule R L M) :
+  (maximal_trivial_hom f m : N) = f m :=
+rfl
+
+/-- The maximal trivial submodules of Lie-equivalent Lie modules are Lie-equivalent. -/
+def maximal_trivial_equiv (e : M ≃ₗ⁅R,L⁆ N) :
+  maximal_trivial_submodule R L M ≃ₗ⁅R,L⁆ maximal_trivial_submodule R L N :=
+{ to_fun    := maximal_trivial_hom (e : M →ₗ⁅R,L⁆ N),
+  inv_fun   := maximal_trivial_hom (e.symm : N →ₗ⁅R,L⁆ M),
+  left_inv  := λ m, by { ext, simp, },
+  right_inv := λ n, by { ext, simp, },
+  .. maximal_trivial_hom (e : M →ₗ⁅R,L⁆ N), }
+
+@[simp] lemma maximal_trivial_equiv_symm_eq_symm (e : M ≃ₗ⁅R,L⁆ N) :
+  (maximal_trivial_equiv e).symm = maximal_trivial_equiv e.symm :=
+rfl
```

Comment:
Can you add a `simp` lemma along the lines of  `maximal_trivial_equiv lie_equiv.refl = lie_equiv.refl` too?